website/docs/sql/scalar_func/arrays.md

```diff
@@ -0,0 +1,53 @@
+<code>**array_ndims(**array**)**</code> → `int`| Returns the numbers of dimensions in the array (always `1`).| `array_ndims(array[1,2,3])` → `1`
+<code>**array_lower(**array**)**</code> → `int` | Returns the index of the first element in the array (always `1`). |`array_lower(array[1,2,3])` → `1`
+<code>**array_upper(**array**)**</code> → `int` | Returns the index of the last element in the array (always equal to `array_length(array)`). |`array_upper(array[1,2,3])` → `3`
+<code>**array_to_string(**array, text [, text]**)**</code>| Converts the array into a textual representation, with the given element separator and (optional) null indicator. | `array_to_string(array[1,2,3], ';')` → `1;2;3`<br/>`array_to_string(array[3,2,1,null], '⏰', '🎉')` → `3⏰2⏰1⏰🎉` 
+
+## Transformations
+The following functions produce new array values from existing ones.
+These operations are compatible with all (nullable and non-nullable) element types.
+
+Signature|Description|Example
+---|---|---
+<code>**array_prepend(**T,array(T)**)**</code> → `array(T)`<br/>or<br/><code>T **\|\|** array(T)</code> → `array(T)`| Inserts a value at the front of the array.| `array_prepend(1, array[2,3,4])` → `array[1,2,3,4]` 
+<code>**array_append(**array(T), T**)**</code> → `array(T)`<br/>or<br/><code>array(T) **\|\|** T</code> → `array(T)`| Inserts a value at the end of the array.| `array_append(array[1,2,3], 4)` → `array[1,2,3,4]` 
+<code>**array_concat(**array(T), array(T)**)**</code> → `array(T)`<br/>or<br/><code>array(T) **\|\|** array(T)</code> → `array(T)`| Concatenates two arrays. | `array_concat(array[1,2], array[3,4])` → `array[1,2,3,4]`  
+
+
+## Inner Products
+Hyper offers high-performance implementations of common inner products in vector spaces. These functions operate on two input vectors (represented as arrays) and produce a scalar value of type `double precision`. 
+
+Contrary to other array operations, vector functions only apply to a certain set of arrays:
+ - the element type must be `real not null` or `double precision not null`
+ - both arrays must have the same element type
+ - both arrays must have the same length
+
+In the following table, signatures and examples are abbreviated for clarity. `vec` denotes a suitable array type (i.e., `array(real not null)` or `array(double precision not null)`). It is implied that both array arguments have the same element type. Similarly, the `{1.0, 2.0, 3.0}` syntax in the example column represents a suitable array value (e.g., `'{1.0, 2.0, 3.0}'::array(real not null)`).
+
+|Signature|Description|Example
+|---|---|---|
+|<code>**dot_product(**vec, vec**)**</code> → `double precision`| Computes the conventional [dot product][dot-product]  between two vectors. | <code>dot_product({1.0, 2.0, 3.0}, {-1.0, 2.0, -3.0})</code> →  `-6.0`
```

Comment:
We should add math definitions here.

Reply:
We would need to enable [KaTeX support](https://docusaurus.io/docs/markdown-features/math-equations) first. Is this worth it, though? The linked Wikipedia articles should provide plenty of context.